src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -194,6 +194,7 @@ public static JaegerTag ToJaegerTag(this KeyValuePair<string, object> attribute)
                 int i => new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: Convert.ToInt64(i)),
                 long l => new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: l),
                 float f => new JaegerTag(attribute.Key, JaegerTagType.DOUBLE, vDouble: Convert.ToDouble(f)),
+                short sh => new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: Convert.ToInt64(sh)),
```

Comment:
Number types are hard 😄. Should we be considering byte/sbyte? What about unsigned variants that would otherwise fit in a signed 64 bit integer like UInt16, UInt32?

Reply:
These are good considerations! In my PRs I've always centered on the four "primitives" (double, long, bool, and string) and then int, short, and float in conversions to long and double respectively. These last three have sensible conversions to existing supported types, but I'm not sure if UInts or bytes would be too different as types that a large amount of meaning would be lost in conversion to long or double?

Reply:
I do not think they're too different from the standpoint of the ability to query and filter the data in some backends. Formatting as a number allows the use of relational operators (`<`, `>`, etc) in queries. I'm not super familiar with Jaeger, but I assume data can be queried in this way.

Can you think of a scenario where loosing the meaning associated with the original datatype chosen would be bad? I think that meaning is often lost anyways when data is serialized to formats like JSON.

Other SDKs have avoided this problem all together by codifying the set of types available for attributes. Like Java for instance:

https://github.com/open-telemetry/opentelemetry-java/blob/f5f0e1f63f9ae2cbb2130c587f118f79e382c410/api/all/src/main/java/io/opentelemetry/api/common/AttributeType.java#L12-L21